XenAdmin/TabPages/PhysicalStoragePage.cs

```diff
@@ -307,6 +312,33 @@ private void dataGridViewSr_MouseUp(object sender, MouseEventArgs e)
             }
         }
 
+        private void dataGridViewSr_SortCompare(object sender, DataGridViewSortCompareEventArgs e)
+        {
+            var sr1 = ((SRRow)dataGridViewSr.Rows[e.RowIndex1]).SR;
+            var sr2 = ((SRRow)dataGridViewSr.Rows[e.RowIndex2]).SR;
+
+            if (e.Column.Index == columnUsage.Index)

```

Comment:
We also need to handle it for the Name and Description column, see for example how it is done on the SrStoragePage.

Reply:
I tested with current code and it works fine for other columns. I guess other column won't be handled in this function so the *e.Handled* won't be set to *true* so it will be sorted automatically by the string sorting.
So is it necessary for us to handle all columns in this function?

Reply:
1) If you change the storage names or descriptions to include numbers at the beginning, e.g. `1 NFS storage`, `10 NFS storage`, `2 NFS storage`, the sorting should be 1, 2, 10, but now it is 1, 10, 2. 
2) if you change the storage names or descriptions to be exactly the same, e.g. `NFS storage`, `NFS storage`, the sorting should be according to the opaque_ref. Have you tested this?
The SrStoragePage has the necessary logic, all you need is to transfer it here.

Reply:
No I haven't tested this case.
Thanks for your advise, I will "revise" :)